### PR TITLE
[Filebeat] Sort array fields in generated data

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Currently the array fields in generated data don't have any sorting or consistent ordering.  This PR sorts any field that is a list/array when the generated data files are created.

## Why is it important?

Whenever data files are generated, there are a large number of changes in the commit due to array fields changing order of the items even when the contents didn't actually change.  This makes it very difficult to actually find what changed, if anything.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally
```
cd beats/filebeat
GENERATE=true TESTING_FILEBEAT_MODULES=<any module> mage -v pythonIntegTest
```
The generated data above should always come out the same and the list orders shouldn't change

## Related issues

- 

## Screenshots
image

### CODE REVIEW

File: filebeat/tests/system/test_modules.py [hunk 1]
```diff
@@ -194,6 +194,9 @@ def _test_expected_events(self, test_file, objects):
                 for k, obj in enumerate(objects):
                     objects[k] = self.flatten_object(obj, {}, "")
                     clean_keys(objects[k])
+                    for key in objects[k].keys():
+                        if isinstance(objects[k][key], list) and len(objects[k][key]) > 0 and not isinstance(objects[k][key][0], dict):
```
Comment (leehinman) [inline filebeat/tests/system/test_modules.py:198]: I'm a little worried about the last check to see if the first element is a dict, if any of the items in the list can't be compared to one another with "<" we will get a runtime error.  What about switching to:
```
if isinstance(objects[k][key], list):
    objects[k][key].sort(key=str)
```

That will force string comparison for everything.

What do you think?

**Review by leehinman — CHANGES_REQUESTED** [review]
Thanks, this will make things more consistent.

File: filebeat/tests/system/test_modules.py (hunk 1 above)
Comment (legoguy1000) [reply filebeat/tests/system/test_modules.py:198]: I never thought about forcing everything to a string. That's a really interesting idea. I'll try that tomorrow.

Comment (legoguy1000) [reply filebeat/tests/system/test_modules.py:198]: It seems to be working, but I'm running into an issue that I had before.  When running the test against all the modules `GENERATE=true mage -v pythonIntegTest`, I end up with some errors below.  Any thoughts?
image

Comment (legoguy1000) [reply filebeat/tests/system/test_modules.py:198]: Things work fine when I do modules individually

Comment (leehinman) [reply filebeat/tests/system/test_modules.py]: Weird.  I'm not seeing those errors when I run your PR locally.

anything in the `build/system-tests/run/<failed test name>` directory?  It looks like some documents got left in elasticsearch `expected event.module=kafka but got redis`.

Might need to clean out docker images and re-run the test.

Comment (legoguy1000) [reply filebeat/tests/system/test_modules.py]: Ok, I'll schwack them and start clean

Comment (legoguy1000) [reply filebeat/tests/system/test_modules.py]: still got the same error.  This is what shows in the `output.log` file under `build/system-tests/run/<failed test name>`
```
2021-05-04T21:36:37.226-0200	INFO	instance/beat.go:651	Home path: [/go/src/github.com/elastic/beats/filebeat] Config path: [/go/src/github.com/elastic/beats/filebeat] Data path: [/go/src/github.com/elastic/beats/filebeat/data] Logs path: [/go/src/github.com/elastic/beats/filebeat/logs]
2021-05-04T21:36:37.237-0200	DEBUG	[beat]	instance/beat.go:704	Beat metadata path: /go/src/github.com/elastic/beats/filebeat/data/meta.json
2021-05-04T21:36:37.237-0200	INFO	instance/beat.go:659	Beat ID: 32ac6545-bf91-437e-8aa3-a64e3359883a
2021-05-04T21:36:37.238-0200	INFO	instance/beat.go:407	filebeat stopped.
2021-05-04T21:36:37.238-0200	ERROR	instance/beat.go:963	Exiting: data path already locked by another beat. Please make sure that multiple beats are not sharing the same data path (path.data).
Exiting: data path already locked by another beat. Please make sure that multiple beats are not sharing the same data path (path.data).
/go/src/github.com/elastic/beats/filebeat/filebeat.test -systemTest -e -d * -once -c /go/src/github.com/elastic/beats/filebeat/build/system-tests/run/test_modules.Test.test_fileset_file_076_elasticsearch/filebeat.yml -E setup.ilm.enabled=false -modules=elasticsearch -M elasticsearch.*.enabled=false -M elasticsearch.gc.enabled=true -M elasticsearch.gc.var.input=file -M elasticsearch.gc.var.paths=[/go/src/github.com/elastic/beats/filebeat/module/elasticsearch/gc/test/test.log] -M *.*.input.close_eof=true
```

Comment (leehinman) [reply filebeat/tests/system/test_modules.py]: That looks like filebeat didn't exit before the next test of a module started.

I'm going to approve and see if we get the same errors with CI.

**Review by leehinman — APPROVED** [review]

File: filebeat/tests/system/test_modules.py (hunk 1 above)
Comment (legoguy1000) [reply filebeat/tests/system/test_modules.py]: ok, I still would like to have the generated data files updated so follow on PRs don't have to worry about the unrelated changes to the data files. So if you're able to run all the modules together and update the files, can you push those changes to the branch??

**Comment by legoguy1000** [comment]
@leehinman I was able to run all the filebeat modules successfully.  Ready for CI tests.

**Comment by legoguy1000** [comment]
I ran it multiple times and no change in order of array fields.

**Pull request merged**